File: py-appscript/tags/py-appscript-0.22.0/appscript_3x/sample/appscript/Print_folder_tree.py
#!/usr/bin/env python3

# Prints the sub-folder hierarchy of a given folder as a list of folder names
# indented according to depth.

from appscript import *


def printfoldertree(folder, indent=''):
    """Print a tab-indented list of a folder tree."""
    print(indent + folder.name.get())
    for folder in folder.folders.get():
        printfoldertree(folder, indent + '\t')


printfoldertree(app('Finder').home.folders['Documents'])

File: rb-appscript/tags/rb-appscript-0.6.0/sample/Print_folder_tree.rb
#!/usr/bin/env ruby

# Prints the sub-folder hierarchy of a given folder as a list of folder names indented according to depth.

# Note: if using the appscript gem, rubygems must be required first:
begin; require 'rubygems'; rescue LoadError; end

require "appscript"
include Appscript

def print_folder_tree(folder, indent='')
    puts indent + folder.name.get
    folder.folders.get.each { |folder| print_folder_tree(folder, indent + "\t") }
end

print_folder_tree(app('Finder').home.folders['Documents'])